iocage/lib/Config/Jail/Properties/Addresses.py

```diff
@@ -43,27 +53,35 @@ def __init__(
         ]=None,
         property_name: str="ip4_address"
     ) -> None:
-
-        if config is not None:
-            self.config = config
-
+        self.config = config
         set.__init__(self)
         self.property_name = property_name
 
-    def add(self, value: str, notify: bool=True) -> None:
+    def add(self, value: IPAddressInput, notify: bool=True) -> None:
         """Add an address to the set."""
-        set.add(self, value)
+        set.add(self, self.__parse_address(value))
         if notify:
             self.__notify()
 
-    def remove(self, value: str, notify: bool=True) -> None:
+    def remove(self, value: IPAddressInput, notify: bool=True) -> None:
         """Remove an address from the set."""
-        set.remove(self, value)
+        set.remove(self, self.__parse_address(value))
         if notify:
             self.__notify()
 
     def __notify(self) -> None:
-        self.config.update_special_property(self.property_name)
+        if self.config is not None:
+            self.config.update_special_property(self.property_name)
+
+    def __parse_address(
+        self,
+        address: IPAddressInput
+    ) -> typing.Union[ipaddress.IPv4Interface, ipaddress.IPv6Interface]:
+        if isinstance(address, str) is True:
+            address = address.lower()
+            if address in ["accept_rtadv", "dhcp"]:
+                return address
+        return ipaddress.ip_interface(address)
```

Comment:
i'm confused about this function name… but i looked it up and it makes perfect sense

but what we now don't actually handle is the [ValueError, which ip_interface raises](https://docs.python.org/3/library/ipaddress.html#ipaddress.ip_interface) if it encounters an error.

Reply:
There are three kinds of address-types supported - Host, Net and Interface.

- Host: may not include a netmask
- Net: may only refer to the network address + netmask
- Interface: stores full client address `<IP>/<mask>`